docker/irods_client/tests/test_session_cached_pw.py

```diff
@@ -0,0 +1,17 @@
+import os
+from ibridges import Session
+
+def test_session_from_cached_pw(config, irods_env):
+    # test only for plain irods
+    if not "set_home_perm" in config:
+        session = Session(irods_env_path=os.path.expanduser("~/.irods/irods_environment.json"))
```

Comment:
So, if I understand correctly, this works because you have done the `iinit` on the command line, right? I first thought that `session._write_pam_password()` would do the same, but I guess it's a different kind of procedure from this?

Reply:
That is true. The command `iinit` automatically writes the password to ~/.irods/.irodsA. I still do not understand why our normal session cannot write the password too. I tried it in the docker manually with the same result. This is still to be investigated.

However, we can so far at least test the second option of creating a session.

Reply:
Yes, that makes sense. In the end it would be good to have all three options of course (with pw, iinit, and ibridges_pam_save).

Reply:
Even in the standard setup, the python irods client fails to retrieve the password. I did not manage to figure out why. I think for now I will leave it. We have the tests for the two ways of creating a session which work.